niworkflows/utils/bids.py

```diff
@@ -167,22 +167,18 @@ def collect_data(dataset, participant_label, task=None, echo=None):
     subj_data = {modality: [x.filename for x in layout.get(**query)]
                  for modality, query in queries.items()}
 
-    def _grp_echos(x):
-        if '_echo-' not in x:
-            return x
-        echo = re.search("_echo-\\d*", x).group(0)
-        return x.replace(echo, "_echo-?")
-
-    if subj_data["bold"]:
-        bold_sess = subj_data["bold"]
-
-        if any(['_echo-' in bold for bold in bold_sess]):
-            ses_uids = [list(bold) for _, bold in groupby(bold_sess, key=_grp_echos)]
-            ses_uids = [x[0] if len(x) == 1 else x for x in ses_uids]
-        else:
-            ses_uids = bold_sess
-
-    subj_data.update({"bold": ses_uids})
+    if any(['_echo-' in bold for bold in subj_data['bold']]):
+        # Special case: multi-echo BOLD, grouping echos
+        def _grp_echos(x):
+            if '_echo-' not in x:
+                return x
+            echo = re.search("_echo-\\d*", x).group(0)
+            return x.replace(echo, "_echo-?")
+
+        bold_sess = subj_data['bold']
+        ses_uids = [list(bold) for _, bold in groupby(bold_sess, key=_grp_echos)]
+        ses_uids = [x[0] if len(x) == 1 else x for x in ses_uids]
```

Comment:
```suggestion
        ses_uids = [x[0] if len(x) <= 2 else x for x in ses_uids]
```

Reply:
Is there any possibility of `x == []`?

Reply:
I don't believe so... If lacks an `echo` key-value pair you shouldn't hit this condition. Can you think of a case ?

Reply:
Not obviously, but my brain isn't very engaged right now. Mostly just recognizing the pattern and asking you to check. :-)

Reply:
why would we want to pick only the first element if there are two echos? Maybe I lost something when transferring from fmriprep.